packages/mdc-button/_mixins.scss

```diff
@@ -163,39 +273,62 @@
   @include mdc-rtl-reflexive-property(margin, 8px, -4px);
 }
 
-@mixin mdc-button--outlined_() {
-  border-style: solid;
+@mixin mdc-button--outlined_($query: mdc-feature-all()) {
+  $feat-color: mdc-feature-create-target($query, color);
+  $feat-structure: mdc-feature-create-target($query, structure);
 
-  &:disabled {
-    border-color: $mdc-button-disabled-ink-color;
+  @include mdc-feature-targets($feat-structure) {
+    border-style: solid;
+  }
+
+  @include mdc-feature-targets($feat-color) {
+    &:disabled {
+      border-color: $mdc-button-disabled-ink-color;
+    }
   }
 }
 
-@mixin mdc-button--filled_() {
-  @include mdc-button-horizontal-padding($mdc-button-contained-horizontal-padding);
+@mixin mdc-button--filled_($query: mdc-feature-all()) {
+  $feat-color: mdc-feature-create-target($query, color);
+  $feat-structure: mdc-feature-create-target($query, structure);
 
-  &:disabled {
-    background-color: rgba(mdc-theme-prop-value(on-surface), .12);
-    color: $mdc-button-disabled-ink-color;
+  @include mdc-feature-targets($feat-structure) {
+    @include mdc-button-horizontal-padding($mdc-button-contained-horizontal-padding);
   }
-}
 
-@mixin mdc-button--raised_() {
-  @include mdc-elevation(2);
+  @include mdc-feature-targets($feat-color) {
+    &:disabled {
+      background-color: rgba(mdc-theme-prop-value(on-surface), .12);
+      color: $mdc-button-disabled-ink-color;
+    }
+  }
+}
 
-  transition: mdc-elevation-transition-value();
+@mixin mdc-button--raised_($query: mdc-feature-all()) {
+  $feat-animation: mdc-feature-create-target($query, animation);
+  $feat-color: mdc-feature-create-target($query, color);
 
-  &:hover,
-  &:focus {
-    @include mdc-elevation(4);
+  @include mdc-feature-targets($feat-color) {
+    @include mdc-elevation(2);
   }
 
-  &:active {
-    @include mdc-elevation(8);
+  @include mdc-feature-targets($feat-animation) {
+    transition: mdc-elevation-transition-value();
   }
 
-  &:disabled {
-    @include mdc-elevation(0);
+  @include mdc-feature-targets($feat-color) {
```

Comment:
This is a little weird since the _default_ value that mdc-elevation uses does not depend on any mdc-theme color "slot", but we may want to err on the safe side in case it eventually does.